app/components/support_interface/reasons_for_rejection_dashboard_component.rb

```diff
@@ -0,0 +1,45 @@
+module SupportInterface
+  class ReasonsForRejectionDashboardComponent < ViewComponent::Base
+    def initialize(rejection_reasons)
+      @rejection_reasons = rejection_reasons
+    end
+
+    def current_month_rejection_count(reason)
+      values = @rejection_reasons.find { |h| h['key'] == reason && h['time_period'] == 'this_month' }
+      return 0 if values.nil?
+
+      values['count']
+    end
+
+    def total_rejection_count(reason)
+      current_month_rejection_count(reason) + previous_rejection_count(reason)
+    end
+
+    def percentage_rejected_for_reason(reason)
+      count = current_month_rejection_count(reason) + previous_rejection_count(reason)
+      total = @rejection_reasons.inject(0) {|sum, hash| sum + hash['count']}
```

Comment:
@davidgisbey We came a cross a problem with the previous way of doing this when we were updating the spec. Essentially, by running `ApplicationChoice.where.not(structured_rejection_reasons: nil).count` this just counts the number of db records with structured reasons. However, each db record can contain multiple reasons as its stored in JSON. Therefore, it made sense to revert to querying the splat and totalling these values.